sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -3921,7 +3921,7 @@ slm_gather(simd<uint32_t, N / VS> byte_offsets, simd_mask<N / VS> mask,
   static_assert(Alignment >= sizeof(T),
                 "slm_gather() requires at least element-size alignment");
 
-  if constexpr (VS > 1 || (!detail::isPowerOf2(N, 32) &&
+  if constexpr (VS > 1 || (!(detail::isPowerOf2(N, 32) && sizeof(T) <= 4) &&
```

Comment:
So today it looks like we will go into this if statement for `sizeof(T) == 8` data, and end up callling `__esimd_lsc_load_merge_slm`

With this change we will go into the final else and call `__esimd_gather_masked_scaled2` right? Is that correct?

If so, can you explain why we want to call this intrinsic instead of the other? The current one seems to be the LSC one which I would expect to be required for 64-bit data.

Reply:
The change here looks good. Please add a test case for 64-bit types to slm_gather.cpp (the one that is runned with new LLVM IR available).

Reply:
Ah I see, thanks

Reply:
bit-cast to 32-bits + 32-bit slm_scatter can be used to initialize SLM.

Reply:
Ok, E2E test can wait for slm_scatter to support 8-bytes, but please create compile-time only test-case(s) in memory_properties.cpp test